packages/opentelemetry-metrics/src/export/aggregators/histogram.ts

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Aggregator, Point } from '../types';
+import { HrTime } from '@opentelemetry/api';
+import { hrTime } from '@opentelemetry/core';
+
+export type Checkpoint = {
+  buckets: {
+    boundaries: number[];
+    counts: number[];
+  };
+  sum: number;
+  count: number;
+};
+
+/** Basic aggregator which calculates a Sum from individual measurements. */
+export class HistogramAggregator implements Aggregator {
+  private _lastCheckpoint: Checkpoint;
+  private _currentCheckpoint: Checkpoint;
+  private _lastCheckpointTime: HrTime = [0, 0];
+  private _boundaries: number[];
+
+  constructor(boundaries: number[]) {
+    if (boundaries === undefined || boundaries.length === 0) {
+      throw new Error(`HistogramAggregator should be created with boundaries.`);
```

Comment:
(Thinking out loud) do you think we should define some default bucket boundaries instead of throwing an exception?

This is what I saw in prom-client: https://github.com/siimon/prom-client/blob/master/lib/histogram.js#L15

Reply:
That would works if you expect the histogram to contain a specific measure (like latency) but i don't think that would be a good idea, user should define it themselves depending on what they are trying to measure.

Reply:
> That would works if you expect the histogram to contain a specific measure (like latency) but i don't think that would be a good idea ...

Hmm that's true. I had this issue in mind, but still added the comment :)